File: 0169-majority-element/0169-majority-element.go
func majorityElement(nums []int) int {
    m := make(map[int]int)

    for _, n := range nums {
        m[n] += 1
    }

    var ans, cnt = 0, 0
    for i, val := range m {
        if cnt < val {
            cnt = val
            ans = i
        }
    }

    return ans
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        cnt = 0
        output = 0
        for num in nums:
            if num == output:
                cnt += 1
            else:
                if cnt < 1:
                    output = num
                else:
                    cnt -= 1
        return output
        